cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -1129,19 +1129,20 @@ private BoolQueryBuilder createDateQuery(QueryTerm queryTerm, @Nullable String f
    */
   private QueryBuilder createDateRangeQuery(QueryTerm queryTerm, String field) {
     Long date = queryTerm.getDate();
+    Long oneDay = TimeUnit.DAYS.toMillis(1);
     switch (queryTerm.getComparison()) {
-      // +/- 1d changes the date by 24 hours (1 whole day).
+      // +/- oneDay changes the date by 24 hours (1 whole day).
       // rangeQuery must therefore be gte/lte instead of gt/lt to not exclude midnight dates by accident.
       case GREATER:
-        return QueryBuilders.rangeQuery(field).gte(date + 1d);
+        return QueryBuilders.rangeQuery(field).gte(date + oneDay);
       case GREATER_OR_EQUAL:
         return QueryBuilders.rangeQuery(field).gte(date);
       case LESS:
-        return QueryBuilders.rangeQuery(field).lte(date - 1d);
+        return QueryBuilders.rangeQuery(field).lte(date - oneDay);
       case LESS_OR_EQUAL:
-        return QueryBuilders.rangeQuery(field).lte(date);
+        return QueryBuilders.rangeQuery(field).lt(date + oneDay);
```

Comment:
why is this change needed?

Reply:
Once again for the 24h issue, so if creation time is 2019-08-16 at 8am, and we want dates less than or equal to 2019-08-16 then we will still want to return something created on 2019-08-16 at 8am. hope that makes sense!